components/org.wso2.carbon.micro.integrator.extensions/org.wso2.carbon.micro.integrator.management.apis/src/main/java/org/wso2/carbon/micro/integrator/management/apis/TemplateResource.java

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.micro.integrator.management.apis;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.config.SynapseConfiguration;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.endpoints.Template;
+import org.apache.synapse.mediators.template.TemplateMediator;
+import org.json.JSONArray;
+import org.json.JSONObject;
+import org.wso2.carbon.inbound.endpoint.internal.http.api.APIResource;
+
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * Represents template resources defined in the synapse configuration.
+ **/
+public class TemplateResource extends APIResource {
+
+    private static final Log log = LogFactory.getLog(TemplateResource.class);
+
+    /* Name of the JSON list of endpoint templates */
+    private static final String ENDPOINT_TEMPLATE_LIST = "endpointTemplateList";
+    /* Name of the JSON list of sequence templates */
+    private static final String SEQUENCE_TEMPLATE_LIST = "sequenceTemplateList";
+    /* Template type parameter */
+    private static final String TEMPLATE_TYPE_PARAM = "templateType";
+    /* Template name parameter */
+    private static final String TEMPLATE_NAME_PARAM = "templateName";
+    /* Possible values for TEMPLATE_TYPE_PARAM  */
+    private static final String ENDPOINT_TEMPLATE_TYPE = "endpoint";
+    private static final String SEQUENCE_TEMPLATE_TYPE = "sequence";
+
+    public TemplateResource(String urlTemplate) {
+        super(urlTemplate);
+    }
+
+    @Override
+    public Set<String> getMethods() {
+        Set<String> methods = new HashSet<>();
+        methods.add("GET");
+        methods.add("POST");
```

Comment:
How are we using the POST methods with these Management API , could you please give an example .

Reply:
This should be get only. Will remove the post.

Reply:
fixed